torch/_inductor/codegen/cuda/cuda_scheduling.py

```diff
@@ -1,10 +1,80 @@
+from typing import cast, List
+
 from ... import config
 from ...codecache import code_hash, get_path
+from ...scheduler import FusedSchedulerNode, Scheduler, SchedulerNode
 from ...utils import get_fused_kernel_name, get_kernel_metadata
 from ...virtualized import V
-
 from ..common import IndentedBuffer
 from ..triton import TritonScheduling
+from .cuda_kernel import CUDATemplateBuffer
+
+
+class CUDASchedulerNode(SchedulerNode):
+    def __init__(self, scheduler: Scheduler, node: CUDATemplateBuffer, group_fn):
+        assert isinstance(node, CUDATemplateBuffer)
+        super().__init__(scheduler, node, group_fn)
+        self.group_fn = group_fn  # keeping this to enable cloning during fuse_epilogue
+
+    def can_fuse_epilogue(self, other_node: SchedulerNode) -> bool:
+        cnode: CUDATemplateBuffer = cast(CUDATemplateBuffer, self.node)
+        if other_node.get_device() != self.get_device():
+            return False
+        return cnode.can_fuse_epilogue(other_node.node)
+
+    def fuse_epilogue(self, other_node: SchedulerNode) -> FusedSchedulerNode:
+        assert self.can_fuse_epilogue(other_node)
+        epilogue_nodes = [other_node]
+
+        cnode: CUDATemplateBuffer = cast(CUDATemplateBuffer, self.node)
+        fused_template_buffer = cnode.create_fused_buffer(
+            other_node.node, self.scheduler
+        )
+        self.node = fused_template_buffer
+        fused_node = FusedCUDASchedulerNode(self, self.scheduler, epilogue_nodes)
+        return fused_node
+
+
+class FusedCUDASchedulerNode(FusedSchedulerNode):
```

Comment:
SchedulerNode, which wraps a CUDATemplateBuffer which contains fused nodes. Again, this class might appear unnecessary, but we need a class deriving from FusedSchedulerNode or the Scheduler would break.